Calendar/Calendar.js

```diff
@@ -0,0 +1,292 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { View, Text, StyleSheet, TouchableOpacity, Image } from 'react-native';
+import moment from 'moment';
+import Back from './Back.png';
+
+type Props = {
+  onChange: () => void,
+  selected?: string,
+  calPadding?: number,
+  calWidth?: number,
+  calendarDateFormat?: string,
+  defaultView?: string,
+  close?: () => void,
+}
+
+type State = {
+  currentDate: string,
+  activeView: string,
+}
+
+const BRAND_COLOR = '#ffd400';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'white',
+    borderWidth: 1,
+    borderRadius: 5,
+  },
+  headerStyle: { fontSize: 20, fontWeight: 'bold' },
+  headerDayMonth: { paddingRight: 10 },
+  calRow: { flexDirection: 'row', justifyContent: 'space-between' },
+  centerItems: { alignItems: 'center', justifyContent: 'center' },
+  calCell: { flex: 1, alignItems: 'center', justifyContent: 'center' },
+  isSelectedContainer: { borderRadius: 5, backgroundColor: BRAND_COLOR },
+  isSelectedText: { color: 'white' },
+});
+
+const calendarData = {
+  toggleDate: {
+    day: 1,
+    month: 12,
+    year: 12,
+  },
+  formats: {
+    day: 'M',
+    month: 'M',
+    year: 'y',
+  },
+  header: {
+    month: 'MMM',
+    year: 'YYYY',
+  },
+};
+class Calendar extends PureComponent<Props, State> {
+  static defaultProps = {
+    calWidth: 300,
+    calPadding: 5,
+    selected: '',
+    calendarDateFormat: 'YYYY-MM-DD',
+    defaultView: 'day',
+    close: () => {},
+  };
+
+  state = {
+    activeView: this.props.defaultView,
+    currentDate: this.props.selected
```

Comment:
currentDate is not needed on state. Use `value` props directly in all places

Reply:
used currentDate for all views.
used moment